### ISSUE TITLE
[Bug]: Unable to estimate funds (bullshit message that the transaction will fail) when account has insufficient funds

### ISSUE THREAD

**Issue by php4fan** [MetaMask/metamask-extension#14986]
### Describe the bug

This is unbelievable. I lost a whole afternoon trying to figure out what I was doing wrong (which was nothing) because of this stupid bug.

### Steps to reproduce

1. Connect to the BNB Smart Chain network
2. Select an account where you have no funds, or less than 0.002 BNB.
3. Try to send this transaction:

```
To:
0xf4BE4f05c35A82BD2369c0bFdcf3588dE07967aC

Value:
0.002 BNB or more

Hex data:
0x6f833aea000000000000000000000000000000000000000000000000000000000000001c25990dd2e2754b2a11f68dc3d3d5811c7c6bb578ec383dd2eac228ef9035ce63188ecf89966b4573dcf7226f1dfede2d055889d245db9b654b549ffcd17e8f7700000000000000000000000000000000000000000000000000000000000002330000000000000000000000000000000000000000000000000000000062ae22ea
```
**DO NOT send the transaction if you have enough funds**

This transaction does NOT fail if you have enough funds (but do NOT do it with enough funds, because if it succeeds, then any subsequent tries with the same data will fail, and therefore you will be unable to test the issue anymore).

To reproduce the issue, do as per the steps above, with an account that does not have funds.

Expected behavior
===
Since this is a transaction that would succeed given enough funds, MetaMask should **estimate gas fees correctly**, and only show the **"insufficient funds"** error.

Observed behavior
===
This nonsense:
![image](https://user-images.githubusercontent.com/13969584/174454620-289c6b95-ce75-4e5a-898a-327c37cfb535.png)

This is what you normally get when you are trying to send a transaction that would fail because of errors thrown by the contract. But that is not the case here.

### Version

10.14.6

### Browser

Chrome

### Operating system

Linux

**Comment by seaona** [comment]
Thank you for your report @php4fan. That's interesting, as I cannot proceed to the Confirmation Screen without funds (Next button is disabled). Could you clarify how did you actually get into the Confirmation screen with an empty account? Ty in advance!

![image](https://user-images.githubusercontent.com/54408225/174580152-921d74f2-8b05-4d73-9d86-cc21911fb042.png)

**Comment by php4fan** [comment]
Hmm, I'm not sure, I get the same as you now. Actually I did have some funds but not enough for the transaction. Perhaps I had just barely over 0.002, just barely enough for a normal transfer of 0.002 but not for that particular transaction, or something like that...

That however rises another issue: whether or not I have enough funds, I should be able to get the gas fee estimation, which also tells me if the transaction is expected to succeed or fail.

**Comment by bschorchit** [comment]
This (when merged) should solve the seeing gas fee estimation part: https://github.com/MetaMask/metamask-extension/pull/14634

**Comment by php4fan** [comment]
I think I've found out how I got the issue and how to systematically reproduce (and how I reproduced it the first time):

0. Be connected to the BNB Smart Chain with an account that has less than `0.002` BNB.
1. Visit this page: https://opentarget.org/test/stupid/mm/trans/
2. Ignore what it says. **Click on the "Send transaction now"** button (if there's no such button something is wrong)

I get the same behavior as in the original report. This is a transaction that will not fail if you have enough funds.

If I do the same with an account that has **enough funds, it works**.

**_Do not complete the transaction_**. Even if you were willing to through away a few pennies for the sake of testing, if this transaction is succesfully completed once, attempting to do it again will be expected to fail, and hence it will be impossible to test the issue with the same transaction.

Sorry that I don't have time to create a more minimal example.
